lib/cmsis_rtos_v2/thread.c

```diff
@@ -83,36 +100,69 @@ osThreadId_t osThreadNew(osThreadFunc_t threadfunc, void *arg,
 	s32_t prio;
 	struct cv2_thread *tid;
 	static u32_t one_time;
+	size_t stack_size;
+	u32_t this_thread_num;
+	u32_t this_thread_num_dynamic;
+
+	if (k_is_in_isr()) {
+		return NULL;
+	}
+
+	if (thread_num >= CONFIG_CMSIS_V2_THREAD_MAX_COUNT) {
```

Comment:
The purpose of this test is that the __ASSERT only fails on a system that have those enabled, without this check and returning NULL more threads than the maximum can be created. I'd say it's safer to have the always-enabled runtime check than relying on the __ASSERT alone. What do you think?

Reply:
Yes, you are correct. I use ASSERTs so much that i have taken them for granted. :(
Since this is a condition which essentially means there is no point in proceeding further, i believe we should use only the if statement and remove __ASSERT. Is it possible for you to scrub the source code of CMSIS RTOS v2 for such usages and correct it?